crates/task-impls/src/rewind.rs

```diff
@@ -0,0 +1,81 @@
+use std::collections::VecDeque;
+use std::sync::Arc;
+
+use anyhow::Result;
+use async_broadcast::{Receiver, Sender};
+use async_trait::async_trait;
+use hotshot_task::task::TaskState;
+use hotshot_types::traits::node_implementation::NodeType;
+use std::fs::OpenOptions;
+use std::io::Write;
+
+use crate::events::HotShotEvent;
+
+#[cfg(not(any(debug_assertions, test)))]
+const REWIND_MAX_DEPTH: usize = 1_000;
```

Comment:
Is this depth too deep for production? I think memory wise it's okay, as, even if this was full of our absolutely largest event, we'd still be under 1mb of total memory usage.

Reply:
No this should be totally fine memory wise, I'd actually be more worried it is too small.